Notebooks/word2vec/Load_Text_Set.py

```
@@ -0,0 +1,59 @@
+# *******************************************************************************
+# **
+# **  Author: Michael Lomnitz (mrlomnitz@lbl.gov)                               
+# **  Python module used to download and load text8 dataset for embedding module
+# **  un Udacity course
+# **
+# *******************************************************************************
+import os
+import zipfile
+import collections
+import tensorflow as tf
+from urllib.request import urlretrieve
+
+url = 'http://mattmahoney.net/dc/'
+expected_bits = 31344016
+
+# Read data from a downloaded .zip
+def read_data(file):
+    with zipfile.ZipFile(file) as f:
+        data = tf.compat.as_str(f.read(f.namelist()[0])).split()
+    return data
+
+# Data object
+class text_8(object):
+    def __init__(self,  vocabulary = 200000):
+        filename  = 'text8.zip'
+        # Init the data set, if file is absent download it
+        if not os.path.exists(filename):
+            print('File is missing, retrieving ', url+filename)
+            filename, _ = urlretrieve(url+filename,filename)
+        # Verify that datasert was correctly loaded by checking bits.
+        if os.stat(filename).st_size != expected_bits:
+            print('Cannot load & verify. Please sort this out')
```

This should raise an exception
https://docs.python.org/3/tutorial/errors.html